app/controllers/posts_controller.rb

```diff
@@ -28,6 +30,20 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+
+    # Delete all likes associated with the post
+    @post.likes.destroy_all
+
+    if @post.destroy
+      redirect_to user_posts_path(current_user), notice: 'Post deleted successfully'
+    else
+      flash.new[:alert] = @post.errors.full_messages.first if @post.errors.any?
+      render :show, status: 400
+    end
+  end
+
```

Comment:
- [ ] Good job so far, When I delete a post that has no likes and comments the post is successfully deleted, or if the post has likes but no comments the post is successfully deleted but if the post has comments when I try to delete it I get an error. You should add `@post.comments.destroy_all` so that even if the post has comments, when you delete the post the comments can be destroyed too and the post will be deleted successfully.

<img width="815" alt="Screenshot 2023-09-07 at 14 55 47" src="https://github.com/iamsjunaid/quill-quest/assets/74501523/2104d0b2-21f2-4b74-abe5-548a011cdae0">

Reply:
Hi @cynthiainga 👋, Thanks for the thorough review🙂!! I have implemented the logic to delete all associated comments when deleting a post.